cmd/crates/stellar-ledger/src/lib.rs

```diff
@@ -304,12 +301,19 @@ fn test_network_hash() -> Hash {
     use sha2::Digest;
     Hash(sha2::Sha256::digest(TEST_NETWORK_PASSPHRASE).into())
 }
+
 #[cfg(test)]
 mod test {
```

Comment:
These tests don't need anything non-public, but I kind of like the idea of leaving these unit tests in `src/lib.rs` - I like the separation of internal:unit vs external:integration like is mentioned in that article you shared. And then we also have a way to run some quick tests via `cargo run --lib` which avoids the longer running integration tests in `tests/`. 

The only thing weird is that they reach out to `tests/test/emulator_http_transport.rs`. That is just so there is something that implements the `Exchange` trait to pass into the `LedgerSigner` initializer. I am going to leave this for now... but this could also be mocked out pretty easily too in the future.